actionpack/test/dispatch/cookies_test.rb

```diff
@@ -1300,6 +1300,22 @@ def test_vanilla_cookie_with_expires_set_relatively
     end
   end
 
+  def test_purpose_metadata_for_signed_cookies
+   cookies.signed[:user_id] = 50
+   cookies.signed[:discount_percentage] = 10
+
+   cookies[:discount] = cookies[:user_id]
+   assert_nil cookies.signed[:discount_percentage]
+  end
+
+  def test_purpose_metadata_for_encrypted_cookies
+    cookies.encrypted[:user_id] = 50
+    cookies.encrypted[:discount_percentage] = 10
+
+    cookies[:discount] = cookies[:user_id]
+    assert_nil cookies.encrypted[:discount_percentage]
+  end
+
```

Comment:
The tests related to purpose and expiry metadata are failing, because I can't switch on - `ActionDispatch::Cookies::AbstractCookieJar.use_cookies_with_metadata` which is set to false by default.
I tried turning it to true in the metadata tests like this -
```
  def test_purpose_metadata_for_encrypted_cookies
    ActionDispatch::Cookies::AbstractCookieJar.use_cookies_with_metadata = true
    cookies.encrypted[:user_id] = 50
    cookies.encrypted[:discount_percentage] = 10

    cookies[:discount] = cookies[:user_id]
    assert_nil cookies.encrypted[:discount_percentage]
    ActionDispatch::Cookies::AbstractCookieJar.use_cookies_with_metadata = false
  end
```

I also tried creating a subclass of CookiesTest with `setup` and `teardown` methods. But it screws up everthing.